File: java/com/aa/act/interview/org/Name.java
package com.aa.act.interview.org;

public class Name {

	private String first;
	private String last;
	
	public Name(String first, String last) {
		if(first == null)
			throw new IllegalArgumentException("first name cannot be null");
		if(last == null)
			throw new IllegalArgumentException("last name cannot be null");
		this.first = first;
		this.last = last;
	}
	
	public String getFirst() {
		return first;
	}
	
	public String getLast() {
		return last;
	}

	@Override
	public String toString() {
		return first + " " + last;
	}
}

File: javascript-es5/Name.js
module.exports = function (first, last) {
  if (!first) {
    throw new Error('first name cannot be null');
  }
  if (!last) {
    throw new Error('last name cannot be null');
  }
  return {
    getFirst: function () { return first; },
    getLast: function () { return last; },
    toString: function () { return `${first} ${last}`; },
  };
};
